shared/src/utils/validators/is-string-literal-union.ts

```diff
@@ -0,0 +1,27 @@
+import type { ValidationOptions, ValidationArguments } from 'class-validator';
+import { makeValidator } from './make-validator';
+
+export type AllowedValues<T extends string> = { [key in T]: true };
```

Comment:
This is an excellent idea. One problem I saw, was that the additional `...allowedValues` variable on its own didn't make any sense. But because they now have an explicit type, we can add a comment to this type describing what their purpose is.

So something like this, maybe?

```suggestion
/**
 * A type for validating unions of string literals.
 *
 * @example
 * ````ts
 * type StringLiteralUnion = 'a' | 'b';
 *
 * const stringLiteralUnionAllowedValues: AllowedValues<StringLiteralUnion> = {
 *     a: true,
 *     b: true,
 * };
 *
 * class MyClassToValidate {
 *     @IsStringLiteralUnion<StringLiteralUnion>(stringLiteralUnionAllowedValues)
 *     public readonly myString: StringLiteralUnion;
 * }
 *
 * ````
 */
export type AllowedValues<T extends string> = { [key in T]: true };
```